watchgod/watcher.py

```diff
@@ -25,19 +25,29 @@ def should_watch_dir(self, entry):
     def should_watch_file(self, entry):
         return True
 
-    def _walk(self, dir_path, changes, new_files):
+    def _walk(self, path, changes, new_files):
+        if os.path.isfile(path):
+            self._watch_file(path, changes, new_files)
+        else:
+            self._walk_dir(path, changes, new_files)
+
+    def _watch_file(self, path, changes, new_files, mtime=None):
```

Comment:
I'm not sure we need `mtime` argument here. the `mtime = os.stat(path).st_mtime` should always work and be no less performant.